src/Microsoft.AspNetCore.Routing/DefaultInlineConstraintResolver.cs

```diff
@@ -82,7 +82,16 @@ private static IRouteConstraint CreateConstraint(Type constraintType, string arg
             // No arguments - call the default constructor
             if (argumentString == null)
             {
-                return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                try
+                {
+                    return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidRouteException(
```

Comment:
@Eilon - are you ok with changing the type of exception thrown here for 1.1.0?

We need to catch this in MVC, and don't want to catch everything.